SeQuant/core/expressions/expr.cpp

```diff
@@ -369,19 +371,25 @@ ExprPtr Sum::canonicalize_impl(bool multipass, CanonicalizeOptions opts) {
     std::wcout << "Sum::canonicalize_impl: input = "
                << to_latex_align(shared_from_this()) << std::endl;
 
+  // canonicalizing TNs in a sum requires treating named indices as
+  // meaningful/distinct
+  auto opts_copy = opts;
+  opts_copy.ignore_named_index_labels = false;
+
   const auto npasses = multipass ? 3 : 1;
   for (auto pass = 0; pass != npasses; ++pass) {
     // recursively canonicalize summands ...
     // using for_each and directly access to summands
-    sequant::for_each(summands_, [pass, &opts](ExprPtr &summand) {
+    sequant::for_each(summands_, [pass, &opts_copy](ExprPtr &summand) {
       ExprPtr bp;
       const auto rapid = (pass % 2 == 0);
```

Comment:
Only indirectly related to this PR, but do we really still need 2 passes of canonicalization? 🤔